### PR TITLE
Feature/add subst opcode

### PR DESCRIPTION
**Pull request by FiorenzaBusi**
Implemented the new SUBST opcode, based on MUTE13_32 which contains all the analyzed and implemented cases.
Added MUTE and test cases

## Description

Added 2 new mutes to test the new SUBST opcode and MOVE.
In the class` MuteExecutionTest.kt` added 2 new function to test the new mutes.
In `misc.kt` added the logic to retrieve the data necessary for processing the SUBST from the AST grammar.
In `Statements` added the logic on how the SUBST opcode works and inline variable definition.
Added 7 new rpgle to test particular cases of SUBST opcode.
In `InterpreterTest` added 7 new functions to test some specific cases.

## Checklist:
- [X] There are tests regarding this feature
- [X] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [X] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 1]
```diff
@@ -1532,3 +1532,51 @@ data class XFootStmt(
         xfoot(this, interpreter)
     }
 }
+
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt:1535]: Missing javadoc

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 2]
```diff
@@ -1532,3 +1532,51 @@ data class XFootStmt(
...
+    val length: Expression?,
+    val value: Expression,
+    val startPosition: Expression?,
+    val target: AssignableExpression,
+    val operationExtender: String?,
+    override val position: Position? = null
+) : Statement(position) {
+    override fun execute(interpreter: InterpreterCore) {
+
+        val start = if (startPosition?.let { interpreter.eval(it).asString() } != null) {
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt:1547]: Maybe we can improve all tests regarding the "null safety" by using let scoped function and|or the kotlin Elvis operator﻿

**Review by lanarimarco — CHANGES_REQUESTED** [review]
Good work, just some suggestions about your code style which is still java style.
One more thing: also the comments in rpgle tests case should be in english.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt (hunk 2 above)
Comment (FiorenzaBusi) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt]: Changed code using Elvis operator

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 3]
```diff
@@ -1532,3 +1532,59 @@ data class XFootStmt(
...
+            val end = interpreter.eval(it).asString().value.toInt()
+            val endPosition = start + end
+            interpreter.eval(value).asString().value.substring(startIndex = start - 1, endIndex = endPosition - 1)
+        } ?: interpreter.eval(value).asString().value.substring(startIndex = start - 1)
+
+        /** Extended operations on opcode
+         * in case of SUBST replace range of string
+         * in case of SUBST(P) replace string
+         */
+        val newSubstr = if (operationExtender == null) {
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt:1575]: You may use let scoped fuincion and elvis operator also here.

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/evaluation/InterpreterTest.kt [hunk 4]
```diff
@@ -1987,4 +1987,45 @@ Test 6
...
+    fun executeSUBST_00() {
+        assertEquals(listOf("AB"), outputOf("SUBST_00"))
+    }
+
+    @Test
+    fun executeSUBST_01() {
+        assertEquals(listOf("CD"), outputOf("SUBST_01"))
+    }
+
+    // View AST of script
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/evaluation/InterpreterTest.kt:2001]: Remove this comment

**Review by lanarimarco — CHANGES_REQUESTED** [review]
You are almost perfect, just two little things.

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**